front50-web/src/main/groovy/com/netflix/spinnaker/front50/controllers/v2/ApplicationsController.groovy

```diff
@@ -94,7 +94,7 @@ public class ApplicationsController {
     return results
   }
 
-  // TODO(ttomsu): Think through application creation permissions.
+  @PreAuthorize("hasPermission(#app.name, 'APPLICATION', 'CREATE')")
```

Comment:
This we need to feature flag so that we don't have the circular dependency of this depends on a new fiat, but new fiat depends on this

Reply:
This would allow feature flagging on the enforcement of `CREATE`:
```suggestion
  @PreAuthorize("#allowAllApplicationCreate or hasPermission(#app.name, 'APPLICATION', 'CREATE')")
  @ApiOperation(value = "", notes = "Create an application")
  @RequestMapping(method = RequestMethod.POST)
  Application create(@Value('${application.create.allow-all:true}') boolean allowAllApplicationCreate,
                                  @RequestBody final Application app) {
```